cmd/find.go

```diff
@@ -190,142 +199,127 @@ func doFind(clnt Client, ctx *cli.Context) {
-
-			orBool := ctx.Bool("or")
-
-			match := fileContentMatch(fileContent, orBool, ctx)
-
-			if match && ctx.String("print") != "" {
-				doFindPrint(filePath, ctx, fileContent)
-			} else if match {
-				printMsg(findMSG{fileContent})
-			}
-
-			if !ctx.Bool("watch") && match && ctx.String("exec") != "" {
-				doFindExec(ctx, filePath, fileContent)
-			}
+		// Maxdepth can modify the filepath to end as a directory prefix
+		// to be consistent with the find behavior, we wont list directories
+		// so any paths which end with a separator are ignored.
+		if strings.HasSuffix(fileContent.Key, separator) {
+			continue
 		}
 
+		match := doFindMatch(ctx, fileContent)
+		if !ctx.Bool("watch") && match && ctx.String("exec") != "" {
+			doFindExec(ctx, fileContent)
+		} else if match && ctx.String("print") != "" {
+			doFindPrint(ctx, fileContent)
+		} else if match {
+			printMsg(findMessage{fileContent})
+		}
 		if ctx.Bool("watch") {
-			doFindWatch(ctx.Args().Get(0), ctx)
+			doFindWatch(ctx, ctx.Args().Get(0))
 		}
 	}
-
+	return nil
 }
 
-// SubArgsHelper formats the string to remove {} and replace each with the appropriate argument
-func SubArgsHelper(args, path string, fileContent contentMessage) string {
-
+// stringsReplace - formats the string to remove {} and replace each
+// with the appropriate argument
+func stringsReplace(args string, fileContent contentMessage) string {
 	// replace all instances of {}
 	str := args
 	if strings.Contains(str, "{}") {
-		str = strings.Replace(str, "{}", path, -1)
+		str = strings.Replace(str, "{}", fileContent.Key, -1)
+	}
+
+	// replace all instances of {""}
+	if strings.Contains(str, `{""}`) {
+		str = strings.Replace(str, `{""}`, strconv.Quote(fileContent.Key), -1)
 	}
 
 	// replace all instances of {base}
 	if strings.Contains(str, "{base}") {
-		str = strings.Replace(str, "{base}", filepath.Base(path), -1)
+		str = strings.Replace(str, "{base}", filepath.Base(fileContent.Key), -1)
+	}
+
+	// replace all instances of {"base"}
+	if strings.Contains(str, `{"base"}`) {
+		str = strings.Replace(str, `{"base"}`, strconv.Quote(filepath.Base(fileContent.Key)), -1)
 	}
 
 	// replace all instances of {dir}
 	if strings.Contains(str, "{dir}") {
-		str = strings.Replace(str, "{dir}", filepath.Dir(path), -1)
+		str = strings.Replace(str, "{dir}", filepath.Dir(fileContent.Key), -1)
+	}
+
+	// replace all instances of {"dir"}
+	if strings.Contains(str, `{"dir"}`) {
+		str = strings.Replace(str, `{"dir"}`, strconv.Quote(filepath.Dir(fileContent.Key)), -1)
 	}
 
 	// replace all instances of {size}
 	if strings.Contains(str, "{size}") {
-		s := humanize.IBytes(uint64(fileContent.Size))
-		str = strings.Replace(str, "{size}", s, -1)
+		str = strings.Replace(str, "{size}", humanize.IBytes(uint64(fileContent.Size)), -1)
 	}
 
-	if strings.Contains(str, "{url}") {
-		s := GetPurl(path)
-		str = strings.Replace(str, "{url}", s, -1)
+	// replace all instances of {"size"}
+	if strings.Contains(str, `{"size"}`) {
```

Comment:
This check has been already done above, line#285. It looks like we don't need the above if block.